featurebyte/models/online_store_compute_query.py

```diff
@@ -0,0 +1,58 @@
+"""
+AggregationResult document model
+"""
+from __future__ import annotations
+
+from typing import List, Optional
+
+import pymongo
+from pydantic import StrictStr
+
+from featurebyte.models.base import (
+    FeatureByteCatalogBaseDocumentModel,
+    PydanticObjectId,
+    UniqueValuesConstraint,
+)
+
+
+class OnlineStoreComputeQueryModel(FeatureByteCatalogBaseDocumentModel):
+    """
+    OnlineStoreComputeQueryModel document
+
+    This model tracks a query that computes an aggregation result to be stored in an online store
+    table. An aggregation result is the outcome of aggregating a tile table using a user specified
+    feature derivation window.
+
+    When a tile is updated in a scheduled task, the related aggregation results are also updated.
+    That is done in TileScheduleOnlineStore by retrieving all the OnlineStoreComputeQueryModel
+    associated with the aggregation_id and executing the sql queries.
+    """
+
+    tile_id: str
+    aggregation_id: str
+    result_name: str
+    result_type: str
+    sql: str
+    table_name: str
+    serving_names: List[StrictStr]
```

Comment:
Any reason to use `StrictStr` only in `serving_names`?

Reply:
There is no particular reason, I think all of them can be `StrictStr` if that's better. Only `serving_names` is directly provided by users, but that should also been validated elsewhere already.

Reply:
Will fix in a follow up PR.